R/utils.R

```diff
@@ -53,6 +53,30 @@ csv_reader <- function(file, verbose = FALSE, guess_max = 1000, ...) {
   return(tibble(data))
 }
 
+#' Custom JSON reading function
+#'
+#' @description Checks for use of memoise and then uses vroom::vroom.
+#' @param file A URL or filepath to a JSON
+#' @param ... extra parameters to be passed to jsonlite::fromJSON
+#' @inheritParams message_verbose
+#' @return A data table
+#' @importFrom tibble tibble
+#' @importFrom jsonlite fromJSON
+#' @concept utility
+json_reader <- function(file, verbose = FALSE, ...) {
```

Comment:
ideally see a test of this using a saved small json file.

Reply:
no comment

Reply:
I’d misread/misunderstood what this request was for, thinking it was asking for a sample of the Vietnam data (which could be tedious to create) but I think I can make a small JSON file to test loading (which be a sub sample of the Vietnam data after all).